src/bin/nydusd/api_server_glue.rs

```diff
@@ -216,7 +216,22 @@ impl ApiServer {
                 info!("exit daemon by http request");
                 ApiResponsePayload::Empty
             })
-            .map_err(|e| ApiError::DaemonAbnormal(e.into()))?;
+            .map_err(|e| {
+                error!("exit fuse service failed {:}", e);
+                ApiError::DaemonAbnormal(e.into())
+            })?;
+
+        // Ensure both fuse and state machine threads have be terminated thus this
+        // nydusd won't race fuse messages when upgrading.
+        d.wait()
+            .map(|_| {
+                info!("fuse service exited by http request");
+                ApiResponsePayload::Empty
```

Comment:
Is it required?

Reply:
Yes, it is. We need to reply properly to API callers.

Reply:
The return value of `d.wait` seems not to be used, but that's okay.